environmental_data_modules/met_post_processor.py

```diff
@@ -0,0 +1,655 @@
+            indexes_to_drop = indexes_to_drop.append(station_drop_indexes)
+
+        # Finally drop all the data that is unwanted
+        return met_data_in.drop(index=indexes_to_drop)
+
+    def drop_single_daily_measurement_stations(self, met_data_in):
+        # group the data by date, and count the readings per day
+        tempgroups = met_data_in.groupby(['siteID', pd.Grouper(key='date', freq='1D')])
+        data_counts = tempgroups.count()
+
+        # some diagnostic output, if required
+        if self._print_stats:
+            self.print_data_count_stats(data_counts)
+
+        # find the stations and days with single temperature measurements for that day
+        temperature1_index = data_counts[data_counts['temperature']==1].index
+        # find the stations and days with more than a single temperature measurement
+        temperature24_index = data_counts[data_counts['temperature']>1].index
+
+        # get the station ID's for both of these indexes
+        station1_list  = temperature1_index.get_level_values(0).unique().to_list()
+        station24_list = temperature24_index.get_level_values(0).unique().to_list()
+
+        # Identify the stations in list 1 but not list 24 - these will be the stations with only single daily readings
+        #   Note: A few stations which seem to be mixed, with long(ish) periods of single daily readings,
+        #         mixed with longer periods of multiple daily readings will be missed by this rough filter.
+        #         We will be filtering later by total hourly data counts, which will catch any examples of
+        #         this where the data sets are too sparse to be used.
+        station_single_list = [x for x in station1_list if x not in station24_list]
+
+        if self._print_stats:
+            print('single daily measurement stations to drop')
+            print(station_single_list)
+
+        # drop all the stations with only single daily readings
+        indexes_to_drop = pd.Int64Index(data=[],dtype='int64')
+        for station in station_single_list:
+            station_drop_indexes = met_data_in[met_data_in['siteID']==station].index
+            indexes_to_drop = indexes_to_drop.append(station_drop_indexes)
+
+        # drop all the data that is unwanted
+        met_data_reduced = met_data_in.drop(index=indexes_to_drop)
+
+        # print diag output for new dataset
+        if self._print_stats:
+            print('new stats for the reduced data:')
+            temp_groups = met_data_reduced.groupby(['siteID', pd.Grouper(key='date', freq='1D')])
+            data_counts = temp_groups.count()
+            self.print_data_count_stats(data_counts)
+
+        return met_data_reduced
+
+
+    #%% function for getting two lists of stations, one for required site, one for reference sites
+
+    def station_listing(self, met_extracted_data, var_string):
+        #Todo Doug: can this be merged with station_listing() in aurn_post_processor (and put in to post_processor)
```

Comment:
It should be merged with that version of station_listing. Will need to read through and work out which to keep - this can be done later (once this PR has been sorted).